pkg/service/query_test.go

```diff
@@ -84,3 +85,21 @@ func TestQueryServiceLoadBalancerWithIngress(t *testing.T) {
 	assert.Equal(t, intstr.FromInt(16686), svc.Spec.Ports[0].TargetPort)
 	assert.Equal(t, svc.Spec.Type, corev1.ServiceTypeLoadBalancer) // make sure we get a LoadBalancer service
 }
+
+func TestQueryServiceSpecifiedNodePortWithIngress(t *testing.T) {
+	name := "TestQueryServiceSpecifiedNodePortWithIngress"
+	selector := map[string]string{"app": "myapp", "jaeger": name, "jaeger-component": "query"}
+
+	jaeger := v1.NewJaeger(types.NamespacedName{Name: name})
+	jaeger.Spec.Query.ServiceType = corev1.ServiceTypeNodePort
+	jaeger.Spec.Query.NodePort = 32767
+	svc := NewQueryService(jaeger, selector)
+
+	assert.Equal(t, "testqueryservicespecifiednodeportwithingress-query", svc.ObjectMeta.Name)
+	assert.Len(t, svc.Spec.Ports, 1)
+	assert.Equal(t, int32(16686), svc.Spec.Ports[0].Port)
```

Comment:
```suggestion
	assert.Equal(t, 16686, svc.Spec.Ports[0].Port)
```

Reply:
This is will cause the test to fail, since the value of `t` is expected to be an `int32` value. While the values themselves will be equal, the `assert.Equal` statement will fail due to a type mismatch.